homeassistant/components/onkyo/media_player.py

```diff
@@ -236,8 +314,18 @@ def __init__(
         self._source_list = list(sources.values())
         self._source_mapping = sources
         self._reverse_mapping = {value: key for key, value in sources.items()}
+        self._sound_mode = None
+        self._sound_mode_list = list(SOUND_MODE_MAPPING.keys())
+        self._sound_mode_mapping = SOUND_MODE_MAPPING
+        self._sound_mode_reverse_mapping = {
+            subval: key
+            for key, values in SOUND_MODE_MAPPING.items()
+            for subval in values
+        }
```

Comment:
Why create this one per entity instance instead of putting this next to `SOUND_MODE_MAPPING` definition so you only compute it once.

Reply:
This was mainly to not get too many definitions at the start of the file. 
Also creating a definition for `SOUND_MODE_REVERSE_MAPPING` would generate another 65 lines of code.

Reply:
65 lines? You're doing it here in 5. You can just keep those 5 lines but move it outside of the class.

Reply:
Aah, I though you meant to initialize the `SOUND_MODE_REVERSE_MAPPING` in the same way as the `SOUND_MODE_MAPPING `, as in to write the entire dictionary out, but then the other way around